java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageMetadataResult.java

```diff
@@ -42,25 +42,39 @@
     this.message = message;
   }
 
+  public static MessageMetadataResult create(ByteBuffer buffer) {
+    return new MessageMetadataResult(-1, buffer, Message.getRootAsMessage(buffer));
```

Comment:
instead of -1 could you calculate a length from `buffer`?

Reply:
I added it as an explicit parameter.